meta-tedge/classes/cargo-tedge.bbclass

```diff
@@ -9,7 +9,20 @@ inherit cargo
 do_compile[network] = "1"
 CARGO_DISABLE_BITBAKE_VENDORING = "1"
 
+# The function cargo_common_do_patch_paths adds extra Git repositories with `name` and `destsuffix` from `SRC_URI`  
+# to the Cargo configuration as a patch so that they will be built during the do_compile step with Cargo.  
+#  
+# However, we add a repository with services that meet these conditions but have nothing  
+# to do with Rust code, so we skip this postfunction to prevent errors.  
+#  
+# Note: If we plan to compile two Rust repositories in a single recipe in the future, this change  
+# must be reverted, and cargo_common_do_patch_paths must be redesigned. However, for now,  
+# it is simpler to just skip it.  
+python cargo_common_do_patch_paths:prepend () {
```

Comment:
It smells that non cargo projects should not be added to `SRC_URI` in the scope of `cargo` class.

[Here](https://docs.yoctoproject.org/ref-manual/classes.html#cargo) is the description about the cargo class. 
> All you need is to use the [SRC_URI](https://docs.yoctoproject.org/ref-manual/variables.html#term-SRC_URI) variable to point to a source repository which can be built by Cargo, typically one that was created by the cargo new command, containing a Cargo.toml file, a Cargo.lock file and a src subdirectory.

Reply:
Exactly, that's why we skip this patch to prevent it (we don't have two rust repos in the same recipe anyway).